### ISSUE TITLE
Loading pre-trained model and testing on image pairs

### ISSUE THREAD

**Issue by foromer4** [rpautrat/SuperPoint#18]
Hi, 
I'm trying to load the pre-trained model and use it to test inference on some sample images.
I'm having trouble doing so, if you could provide with a sample it would be great. here is what I tried:         ```
with open("./configs/magic-point_coco_export.yaml", 'r') as f:
            config = yaml.load(f)
        dataset = get_dataset(config['data']['name'])(**config['data'])
        model = get_model(config['model']['name'])(
            data=dataset.get_tf_datasets(),
            n_gpus=1, **config['model'], pred=True)

        model.load("/home/omer/devel/git/ext_repos/SuperPoint/superpoint/saved_models/magic-point_17-06-2018")
        model.__enter__()

        model.__exit__()
```

Another attempts:
 ```
 GRAPH_PB_PATH = '/home/omer/devel/git/ext_repos/SuperPoint/superpoint/saved_models/magic-point_17-06-2018/saved_model.pb'
        with tf.Session() as sess:
            print("load graph")
            with gfile.FastGFile(GRAPH_PB_PATH, 'rb') as f:
                graph_def = tf.GraphDef()
            graph_def.ParseFromString(f.read())
            sess.graph.as_default()
            tf.import_graph_def(graph_def, name='')
            graph_nodes = [n for n in graph_def.node]
            names = []
            for t in graph_nodes:
                names.append(t.name)
            print(names)
```

**Comment by rpautrat** [comment]
Hi!

What error message did you get with the first trial? I think that you could have simply `get_model(config['model']['name'])(data={}, n_gpus=1, **config['model'])`.

You can see an example of how to load a model and do a prediction in `superpoint/export_detections.py` (to be used with the config `configs/magic-point_coco_export.yaml` for example).
You can test it for example on COCO with `python superpoint/export_detections.py superpoint/configs/magic-point_coco_export.yaml magic-point_17-06-2018 --export_name pred_magic-point_17-06-2018 --pred_only`. Starting from this, you should be able to tune it to your needs and use a custom dataset as well.

I hope it helps.

**Comment by foromer4** [comment]
thank you for your  quick reply, I'm trying to use the same config, I still get an error: checkpoint directory is empty
when trying to apply  `model.load("/home/omer/devel/git/ext_repos/SuperPoint/superpoint/saved_models/magic-point_17-06-2018")` , will try to reproduce with the expor_detections.py  script , as far as I can see, I'm doing something similar.

**Comment by rpautrat** [comment]
Yes, what you did is very similar. It is weird though to get the "checkpoint directory is empty" if you used the correct path...
Let me know if the `export_detections.py` works.

**Comment by foromer4** [comment]
I did the following:
 1. Extract the saved_model magic-point_17-06-2018.tar.gz 
 2. Copy the saved_model.pb and variables folder to experiments/magic-point_17-06-2018
 3. Run` python superpoint/export_detections.py superpoint/configs/magic-point_coco_export.yaml magic-point_17-06-2018 --export_name pred_magic-point_17-06-2018 --pred_only` .

result is the same:` Checkpoint directory is empty.`

**Comment by Lyserg25** [comment]
I am getting the same error message when I try python `superpoint/export_detections.py superpoint/configs/magic-point_coco_export.yaml magic-point_17-06-2018 --export_name pred_magic-point_17-06-2018 --pred_only`

**Comment by rpautrat** [comment]
Ah sorry my mistake, the trained model is actually a Tensorflow SavedModel, so you can't load it directly with tf.train.Saver.restore.

But you can slightly modify `export_detections.py` to adapt it to this SavedModel:
- Change the line 40 (currently `net.load(str(checkpoint))`) with `tf.saved_model.loader.load(net.sess, [tag_constants.SERVING], str(checkpoint))`
- Add the two imports that are required:
`import tensorflow as tf`
`from tensorflow.python.saved_model import tag_constants`

With this modification it should work (I just tried it and it worked for me at least). I will try to update the ReadMe in the next days to make it clearer.

**Comment by foromer4** [comment]
ok thank you
